Sources/TestingMacros/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+cmake_minimum_required(VERSION 3.28)
+
+if(POLICY CMP0157)
+  cmake_policy(SET CMP0157 NEW)
+endif()
+
+project(TestingMacros
+  LANGUAGES Swift)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
+
+add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+include(FetchContent)
+set(FETCHCONTENT_BASE_DIR ${CMAKE_BINARY_DIR}/_d)
+FetchContent_Declare(SwiftSyntax
+  GIT_REPOSITORY https://github.com/apple/swift-syntax
+  GIT_TAG main)
```

Comment:
We probably don't want to use the main branch here (although I understand if it's temporarily necessary until swift-syntax-600 is officially tagged.)

Reply:
Yes, I'll need guidance on the preference here.  It has to be something later than [this PR](https://github.com/apple/swift-syntax/commit/85af0463ce07b35a49b4abb2e55aeaff5e2a6cc1) which is needed to build properly.  Should I use the commit hash for that diff?  Or leave it `main` with a TODO?  Or other?

Reply:
Unsure. @bnbarham?

Reply:
I'm personally in the camp of pinning to a specific commit hash. It's good from a security perspective. It's good from a build-time/configure-time perspective since CMake won't try to re-fetch once it's pulled it down the first time. If you give it a tag name or a branch, it will try re-fetching each time.

The drawback of course is that someone needs to remember to update it.

Reply:
I too like the revision for the same reasons.  

I note in `Package.swift` we're pulling based on an env variable?
```
from: Version(stringLiteral: Context.environment["SWT_SWIFT_SYNTAX_VERSION"] ?? "600.0.0-latest")
```
Should I do the same thing here?  Check the environment and fallback to the tag?

Reply:
No need—that's mostly for my own at-desk testing of different swift-syntax versions. Now that we've dropped swift-syntax-510 support, we'll likely just delete that.